elastic/kibana
# [Fleet] Only enable output secrets if all Fleet servers are compatible

## Summary

Closes https://github.com/elastic/kibana/issues/173041

Output secret storage requires that all Fleet servers are on version 8.12.0 or above.

The implementation is similar to package policy secrets: https://github.com/elastic/kibana/pull/163627: this PR adds the new `output_secret_storage_requirements_met` flag on the `ingest_manager_settings` saved object.

### Testing

1. Define a preconfigured output wit a secret value in your `kibana.dev.yml` file, e.g.:
   ```yml
   xpack.fleet.outputs:
     - id: my-logstash-output-with-a-secret
       name: preconfigured logstash output with a secret
       type: logstash
       hosts: ['localhost:9999']
       ssl:
         certificate: xxxxxxxxxx
       secrets:
         ssl:
           key: secretLogstashKey
   ```
3. Start ES and Kibana. Do not start a Fleet server.
4. Go to Fleet settings and inspect the preconfigured output: it should have been created and the secret value should not have been set (optionally, you can check in the Console with `GET .fleet-secrets/_search` that the secret was not created). However, a plain text equivalent should have been created (in the example above, `ssl.key` should be set to `secretLogstashKey`).
5. Start a Fleet server on version less than 8.12.0. Kibana should update the output. Again, check that the secret value was not set and that the plain text equivalent is set.
6. Stop the Fleet server and start another one on version 8.12.0 or higher. Kibana should update the output. This time, the secret value should have been set.

### Checklist
